From: MATYJAGUZ075 <matyjaguz075@users.noreply.github.com>
Date: Fri, 30 Oct 2026 00:00:00 +0000
Subject: [PATCH] MediaRouter: always enable A2DP for Bluetooth routes

In some cases MediaRouter skips calling setBluetoothA2dpOn when playback
is not considered active, causing A2DP to not be routed/started even
though the Bluetooth audio route is selected. This results in no audio
on Bluetooth headphones (first connect can timeout/fail) and requires
re-connecting/pairing again to get audio.

Always call setBluetoothA2dpOn(true) when selecting/enabling a
Bluetooth live audio route.
---
diff --git a/media/java/android/media/MediaRouter.java b/media/java/android/media/MediaRouter.java
index xxxxxxx..yyyyyyy 100644
--- a/media/java/android/media/MediaRouter.java
+++ b/media/java/android/media/MediaRouter.java
@@ -1076,10 +1076,9 @@
 
         if (sStatic.isPlaybackActive() && sStatic.mBluetoothA2dpRoute != null
                 && (types & ROUTE_TYPE_LIVE_AUDIO) != 0
-                && (route.isBluetooth() || route.isDefault())) {
+                ) {
             try {
-                sStatic.mMediaRouterService.setBluetoothA2dpOn(sStatic.mClient,
-                        route.isBluetooth());
+                sStatic.mMediaRouterService.setBluetoothA2dpOn(sStatic.mClient, true);
             } catch (RemoteException e) {
                 Log.e(TAG, "Error changing Bluetooth A2DP state", e);
             }
